includes/integrations/divi/scripts/builder-bundle.min.js

```diff
@@ -75,7 +110,41 @@ class ConvertKitDiviProduct extends React.Component {
  * @param 	string 	valueLabel 		Selected value's label.
  * @param 	string 	resourceName 	Resource name (form,product).
  */
-function ConvertKitDiviRenderModule( block, value, valueLabel, resourceName ) {
+function ConvertKitDiviRenderResourceModule( block, value, valueLabel, resourceName ) {
+
+	// Fetch element if no access token or resources exist.
+	const warning = ConvertKitDiviMaybeRenderWarning( block );
+	if ( warning !== true ) {
+		return warning;
+	}
+
+	// Show instructions if no resource selected.
+	if ( value.length === 0 ) {
+		// Return module with text.
+		return React.createElement( 'div', {
+			className: 'convertkit-divi-module'
+		}, 'Configure this module by clicking the settings cog.' );
+	}
+
+	// Return module with text.
+	return React.createElement( 'div', {
+		className: 'convertkit-divi-module'
+	}, 'ConvertKit ' + resourceName + ' "' + valueLabel + '" selected. View on the frontend site to see the ' + resourceName + '.' ); 
+
+}
+
+/**
+ * Returns the Divi module for the given ConvertKit block (form, product, broadcast etc).
+ * if there is no access token or resources.
+ * 
+ * @since 	2.5.7
+ * 
+ * @param 	object 	block 			Block.
+ * @param 	string  value   		Selected value.
+ * @param 	string 	valueLabel 		Selected value's label.
+ * @param 	string 	resourceName 	Resource name (form,product).
+ */
+function ConvertKitDiviMaybeRenderWarning( block ) {
```

Comment:
This is a confusing function - it is named `...RenderWarning` yet if it returns true, it does not render a warning. This makes the callers check if warning is false and then render a warning. You could go two ways to fix this. First, abide by the single responsibility principle and have two functions: one to check the warning and return the type, then another to render it based on that type. Second, return false if there's no warning, and then have your callers just do this `if (warning) return warning`.